# 01 - Dataset Understanding

This notebook is for understanding the structure of the Global Findex 2025 dataset before cleaning, analysis, SQL, or dashboarding.

## Day 2 Goal

- Confirm the dataset can be loaded safely.
- Inspect rows, columns, and data types.
- Understand the unit of analysis.
- Identify useful columns and indicators for the project.
- Avoid cleaning or conclusions at this stage.


## 1. Import Libraries and Set File Path

We start by importing only what is needed. `Path` helps keep the file path readable and safer than typing long strings repeatedly.


In [31]:
from pathlib import Path
import pandas as pd


In [32]:
DATA_PATH = Path(r"D:/financial-inclusion-gap-analysis/data/raw/GlobalFindexDatabase2025.csv")
DATA_PATH.exists()


True

## 2. Load a Small Sample

We load only the first few rows first. This avoids unnecessary memory use and lets us inspect the structure safely.


In [33]:
sample_df = pd.read_csv(DATA_PATH, nrows=5)
sample_df


,countrynewwb,codewb,year,pop_adult,regionwb24_hi,incomegroupwb24,group,group2,account_t_d,fiaccount_t_d,...,con12m_s,con26lm_s,con12w_s,con2f_s,con13_s,con26m_s,con28lm_s,con5a_s,con17c_s,con32h_s
0,Afghanistan,AFG,2011,14575546,South Asia (excluding high income),Low income,all,all,0.090050,0.090050,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,Albania,ALB,2011,2281010,Europe & Central Asia (excluding high income),Upper middle income,all,all,0.282681,0.282681,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,Algeria,DZA,2011,26251587,Middle East & North Africa (excluding high inc...,Lower middle income,all,all,0.332861,0.332861,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,Angola,AGO,2011,12779501,Sub-Saharan Africa (excluding high income),Lower middle income,all,all,0.392035,0.392035,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,Argentina,ARG,2011,30685516,Latin America & Caribbean (excluding high income),Upper middle income,all,all,0.331302,0.331302,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


## 3. Check Columns and Sample Shape

This tells us how many columns are present and gives an early view of the dataset width.


In [34]:
sample_df.shape


(5, 438)

In [35]:
list(sample_df.columns)


['countrynewwb',
 'codewb',
 'year',
 'pop_adult',
 'regionwb24_hi',
 'incomegroupwb24',
 'group',
 'group2',
 'account_t_d',
 'fiaccount_t_d',
 'mobileaccount_t_d',
 'borrow_any_t_d',
 'fin4_d',
 'dig_acc',
 'fin11_2a',
 'fin11a',
 'fin11b',
 'fin11c',
 'fin11f',
 'fin11d',
 'fin11e',
 'fin14a',
 'fin14b',
 'fin14c',
 'fin14d',
 'fin13_1a',
 'fin13_1b',
 'fin26a',
 'fin26b',
 'fin27a',
 'fin27b',
 'fin17f',
 'fin17a_17a1_d',
 'fin17a',
 'fin17b',
 'fin17c',
 'fin22d',
 'fin22e',
 'fin22a_22a1_22g_d',
 'fin22a',
 'fin22a_22g',
 'fin22a_1',
 'fin22b',
 'fin22c',
 'fin24sav',
 'fin24fam',
 'fin24work',
 'fin24bor',
 'fin24sell',
 'fin24other',
 'fin24aVD',
 'fin24aSD',
 'fin24aND',
 'fin24aSD_ND',
 'fin24aP',
 'fin24aN',
 'fin24sav_SD_ND',
 'fin24fam_SD_ND',
 'fin24work_SD_ND',
 'fin24bor_SD_ND',
 'fin24sell_SD_ND',
 'fin24other_SD_ND',
 'fin24sav_VD',
 'fin24fam_VD',
 'fin24work_VD',
 'fin24bor_VD',
 'fin24sell_VD',
 'fin24other_VD',
 'fh1',
 'fin28',
 'fh2',
 'fin29',
 'fin31a_31b',
 '

## 4. Load Context Columns Only

The dataset is wide, so we first inspect the columns that describe each row: country, year, region, income group, and demographic group.


In [36]:
context_columns = [
    "countrynewwb",
    "codewb",
    "year",
    "pop_adult",
    "regionwb24_hi",
    "incomegroupwb24",
    "group",
    "group2",
]

context_df = pd.read_csv(DATA_PATH, usecols=context_columns, low_memory = False)
context_df.head()


,countrynewwb,codewb,year,pop_adult,regionwb24_hi,incomegroupwb24,group,group2
0,Afghanistan,AFG,2011,14575546.0,South Asia (excluding high income),Low income,all,all
1,Albania,ALB,2011,2281010.0,Europe & Central Asia (excluding high income),Upper middle income,all,all
2,Algeria,DZA,2011,26251587.0,Middle East & North Africa (excluding high inc...,Lower middle income,all,all
3,Angola,AGO,2011,12779501.0,Sub-Saharan Africa (excluding high income),Lower middle income,all,all
4,Argentina,ARG,2011,30685516.0,Latin America & Caribbean (excluding high income),Upper middle income,all,all


In [37]:
context_df.shape


(8577, 8)

In [38]:
context_df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8577 entries, 0 to 8576
Data columns (total 8 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   countrynewwb     8577 non-null   object 
 1   codewb           8577 non-null   object 
 2   year             8577 non-null   int64  
 3   pop_adult        7893 non-null   float64
 4   regionwb24_hi    7893 non-null   object 
 5   incomegroupwb24  7893 non-null   object 
 6   group            8577 non-null   object 
 7   group2           8577 non-null   object 
dtypes: float64(1), int64(1), object(6)
memory usage: 536.2+ KB


## 5. Understand Dataset Coverage

We check countries, years, regions, income groups, and group labels. This helps us understand the grain of the dataset.


In [39]:
context_df["year"].value_counts().sort_index()


year
2011    1516
2014    1686
2017    1734
2021    1483
2022     176
2024    1982
Name: count, dtype: int64

In [40]:
context_df["group"].value_counts(dropna=False)


group
gender        1510
age_cat       1510
education     1508
laborforce    1490
income        1484
all            772
urbanicity     303
Name: count, dtype: int64

In [41]:
context_df["group2"].value_counts(dropna=False)


group2
all                      772
men                      755
women                    755
age 25+                  755
ages 15-24               755
secondary edu or more    754
prim edu or less         754
out of laborforce        745
in laborforce            745
richest 60%              742
poorest 40%              742
urban                    152
rural                    151
Name: count, dtype: int64

In [42]:
context_df["regionwb24_hi"].value_counts(dropna=False)


regionwb24_hi
High income                                           2643
Sub-Saharan Africa (excluding high income)            1883
Europe & Central Asia (excluding high income)         1038
Latin America & Caribbean (excluding high income)      930
NaN                                                    684
Middle East & North Africa (excluding high income)     537
East Asia & Pacific (excluding high income)            521
South Asia (excluding high income)                     341
Name: count, dtype: int64

In [43]:
context_df["incomegroupwb24"].value_counts(dropna=False)


incomegroupwb24
High income            2643
Lower middle income    2211
Upper middle income    2089
Low income              950
NaN                     684
Name: count, dtype: int64

## 6. Early Unit of Analysis Check

We test whether a row appears to represent a country-year-demographic group combination. This is not a final conclusion yet; it is an early validation.


In [44]:
possible_key = ["codewb", "year", "group", "group2"]
duplicate_key_count = context_df.duplicated(subset=possible_key).sum()
duplicate_key_count


np.int64(0)

In [45]:
context_df[["countrynewwb", "codewb", "year", "group", "group2"]].head(20)


,countrynewwb,codewb,year,group,group2
0,Afghanistan,AFG,2011,all,all
1,Albania,ALB,2011,all,all
2,Algeria,DZA,2011,all,all
3,Angola,AGO,2011,all,all
4,Argentina,ARG,2011,all,all
5,Armenia,ARM,2011,all,all
6,Australia,AUS,2011,all,all
7,Austria,AUT,2011,all,all
8,Azerbaijan,AZE,2011,all,all
9,Bahrain,BHR,2011,all,all


## 7. Identify Useful Indicator Columns

The dataset contains many indicator columns. We first inspect likely financial inclusion indicators before selecting final columns.


In [46]:
candidate_indicators = [
    "account_t_d",
    "fiaccount_t_d",
    "mobileaccount_t_d",
    "borrow_any_t_d",
    "save_any_t_d",
    "dig_acc",
    "merchant_pay",
    "g20_made",
    "g20_received",
    "inactive_t_d",
]

available_indicators = [col for col in candidate_indicators if col in sample_df.columns]
available_indicators


['account_t_d',
 'fiaccount_t_d',
 'mobileaccount_t_d',
 'borrow_any_t_d',
 'save_any_t_d',
 'dig_acc',
 'merchant_pay',
 'g20_made',
 'g20_received',
 'inactive_t_d']

In [47]:
indicator_preview_cols = context_columns + available_indicators
indicator_preview_df = pd.read_csv(DATA_PATH, usecols=indicator_preview_cols, low_memory=False)
indicator_preview_df.head()


,countrynewwb,codewb,year,pop_adult,regionwb24_hi,incomegroupwb24,group,group2,account_t_d,fiaccount_t_d,mobileaccount_t_d,borrow_any_t_d,dig_acc,inactive_t_d,g20_made,g20_received,merchant_pay,save_any_t_d
0,Afghanistan,AFG,2011,14575546.0,South Asia (excluding high income),Low income,all,all,0.090050,0.090050,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,Albania,ALB,2011,2281010.0,Europe & Central Asia (excluding high income),Upper middle income,all,all,0.282681,0.282681,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,Algeria,DZA,2011,26251587.0,Middle East & North Africa (excluding high inc...,Lower middle income,all,all,0.332861,0.332861,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,Angola,AGO,2011,12779501.0,Sub-Saharan Africa (excluding high income),Lower middle income,all,all,0.392035,0.392035,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,Argentina,ARG,2011,30685516.0,Latin America & Caribbean (excluding high income),Upper middle income,all,all,0.331302,0.331302,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [48]:
indicator_preview_df[available_indicators].describe().T


,count,mean,std,min,25%,50%,75%,max
account_t_d,8487.0,0.608645,0.281292,0.004049,0.373679,0.619084,0.875772,1.000000
fiaccount_t_d,8395.0,0.577825,0.300430,0.004049,0.305506,0.573062,0.874565,1.000000
mobileaccount_t_d,2527.0,0.284001,0.187705,0.000000,0.134864,0.243727,0.406949,0.939388
borrow_any_t_d,6412.0,0.505079,0.147839,0.066738,0.404864,0.493301,0.599242,0.960750
save_any_t_d,6402.0,0.498438,0.180688,0.056150,0.363724,0.486416,0.628206,0.972891
dig_acc,1375.0,0.463432,0.222712,0.028120,0.276928,0.452996,0.630785,0.984625
merchant_pay,1754.0,0.327419,0.220208,0.000000,0.145243,0.266720,0.485721,0.972780
g20_made,5780.0,0.468842,0.276109,0.006904,0.236038,0.413046,0.683106,1.000000
g20_received,5778.0,0.400943,0.233157,0.004907,0.210776,0.344469,0.573565,0.988207
inactive_t_d,869.0,0.078093,0.077275,0.000000,0.018538,0.044762,0.126790,0.409191


## 8. Nepal Quick Check

Because this project may compare Nepal with South Asia and global patterns, we first confirm Nepal exists and inspect its available rows.


In [49]:
nepal_rows = indicator_preview_df[indicator_preview_df["codewb"].eq("NPL")]
nepal_rows.head(20)



,countrynewwb,codewb,year,pop_adult,regionwb24_hi,incomegroupwb24,group,group2,account_t_d,fiaccount_t_d,mobileaccount_t_d,borrow_any_t_d,dig_acc,inactive_t_d,g20_made,g20_received,merchant_pay,save_any_t_d
91,Nepal,NPL,2011,17498633.0,South Asia (excluding high income),Lower middle income,all,all,0.253086,0.253086,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
234,Nepal,NPL,2014,18161615.0,South Asia (excluding high income),Lower middle income,all,all,0.338014,0.338014,0.003360,0.590007,NaN,0.057346,0.056214,0.065320,NaN,0.446701
380,Nepal,NPL,2017,18869412.0,South Asia (excluding high income),Lower middle income,all,all,0.453856,0.453856,NaN,0.660715,NaN,0.112355,0.091124,0.110819,NaN,0.463059
521,Nepal,NPL,2021,20254590.0,South Asia (excluding high income),Lower middle income,all,all,0.540027,0.528021,0.060919,0.539797,NaN,0.140585,0.187359,0.194268,0.054149,0.348317
662,Nepal,NPL,2024,21168213.0,South Asia (excluding high income),Lower middle income,all,all,0.599911,0.575257,0.105834,0.704009,0.176216,0.106261,0.170105,0.212262,0.092866,0.374014
872,Nepal,NPL,2011,17498633.0,South Asia (excluding high income),Lower middle income,gender,men,0.295582,0.295582,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
873,Nepal,NPL,2011,17498633.0,South Asia (excluding high income),Lower middle income,gender,women,0.212196,0.212196,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1146,Nepal,NPL,2014,18161615.0,South Asia (excluding high income),Lower middle income,gender,men,0.367245,0.367245,NaN,0.630162,NaN,NaN,NaN,NaN,NaN,0.456304
1147,Nepal,NPL,2014,18161615.0,South Asia (excluding high income),Lower middle income,gender,women,0.312665,0.312665,NaN,0.555185,NaN,NaN,NaN,NaN,NaN,0.438374
1438,Nepal,NPL,2017,18869412.0,South Asia (excluding high income),Lower middle income,gender,men,0.499813,0.499813,NaN,0.689647,NaN,0.136624,NaN,0.142650,NaN,0.410127


## 9. Day 2 Summary

### What the dataset contains

The dataset contains Global Findex 2025 financial inclusion indicators by country, year, region, income group, and demographic group. It includes indicator related to account owneership, financial institution account, mobile accounts, borrowing, saving, digital payments, and account inactivity.

### Dataset size
- Rows: 8577
- Columns: 438
- Years avaiable: 2011, 2014, 2017, 2021, 2024
- Nepal country code: NPL

### Likely unit of analysis

Each row appears to represent an aggregrated country-year-population-segment observation.

The orginal Global Findex data is based on individual survery responses, but this csv does not show individual survey respondents. Instead, it provides calculated indicators for countries and groups such as all adults, gender groups, income groups, and other population segments.

So, in this project, the unit of analysis should be treated as country/segment-level financial iunclusion analysis.

A practical row meaning is :

`country + year + group + group2`

### Useful context columns identified

- `countrynewwb`
- `codewb`
- `year`
- `pop_adult`
- `regionwb24_hi`
- `incomegroupwb24`
- `group`
- `group2`

### Intial useful indicator columns

- `account_t_d`
- `fiaccount_t_d`
- `mobileaccount_t_d`
- `borrow_any_t_d`
- `save_any_t_d`
- `dig_acc`
- `merchant_pay`
- `g20_made`
- `g20_recieved`
- `inactive_t_d`

### Important descision

This project ahould not be treated as customer-level analysis. It should be treated as country/segment-level financial inclusion analysis using aggregated indicators.

### Next step

Review indicator meanings and select the final indicator that match best match the project questions.